cni-plugin/deployment/linkerd2-cni.yaml

```diff
@@ -0,0 +1,175 @@
+# This ConfigMap is used to configure a self-hosted linkerd2 CNI installation.
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: linkerd2-cni-config
+  namespace: linkerd
+data:
+  incoming_proxy_port: "4143"
+  outgoing_proxy_port: "4140"
+  proxy_uid: "2102"
+  ports_to_redirect: ""
+  inbound_ports_to_ignore: "[4190,4191]"
+  outbound_ports_to_ignore: ""
+  simulate: "false"
+  log_level: "debug"
+  # The CNI network configuration to install on each node. The special
+  # values in this config will be automatically populated.
+  cni_network_config: |-
+    {
+      "name": "linkerd2-cni",
+      "type": "linkerd2-cni",
+      "log_level": "__LOG_LEVEL__",
+      "policy": {
+          "type": "k8s",
+          "k8s_api_root": "https://__KUBERNETES_SERVICE_HOST__:__KUBERNETES_SERVICE_PORT__",
+          "k8s_auth_token": "__SERVICEACCOUNT_TOKEN__"
+      },
+      "kubernetes": {
+          "kubeconfig": "__KUBECONFIG_FILEPATH__"
+      },
+      "linkerd2": {
+        "incoming-proxy-port": __INCOMING_PROXY_PORT__,
+        "outgoing-proxy-port": __OUTGOING_PROXY_PORT__,
+        "proxy-uid": __PROXY_UID__,
+        "ports-to-redirect": __PORTS_TO_REDIRECT__,
+        "inbound-ports-to-ignore": __INBOUND_PORTS_TO_IGNORE__,
+        "outbound-ports-to-ignore": __OUTBOUND_PORTS_TO_IGNORE__,
+        "simulate": __SIMULATE__
+      }
+    }
+---
+# This manifest installs the linkerd2 CNI plugins and network config on
+# each master and worker node in a Kubernetes cluster.
+kind: DaemonSet
+apiVersion: extensions/v1beta1
+metadata:
+  name: linkerd2-cni
```

Comment:
Right, related to your question in the description for this PR, let's call this `linkerd-cni`. The convention in this repo is to use "linkerd" instead of "linkerd2", unless it's easily confused with linkerd1 functionality. And since linkerd1 doesn't have a CNI plugin, you should be good to go. I'd also change the name prefix of all of the other object in this file to `linkerd-`.

Reply:
updated